pkg/api/mlflow/controller/artifact.go

```diff
@@ -26,3 +28,21 @@ func (c Controller) ListArtifacts(ctx *fiber.Ctx) error {
 	log.Debugf("artifactList response: %#v", resp)
 	return ctx.JSON(resp)
 }
+
+// GetArtifact handles `GET /get-artifact` endpoint.
+func (c Controller) GetArtifact(ctx *fiber.Ctx) error {
+	req := request.GetArtifactRequest{}
+	if err := ctx.QueryParser(&req); err != nil {
+		return api.NewBadRequestError(err.Error())
+	}
+	log.Debugf("GetArtifact request: %#v", req)
+
+	artifact, err := c.artifactService.GetArtifact(ctx.Context(), &req)
+	if err != nil {
+		return err
+	}
+	defer artifact.Close()
```

Comment:
I don't like this idea to close `reader` object right here. It will be much more reasonable to close it in a place where actual works does. Controller has to just send bytes without knowing about underlying object.

Reply:
i agree this seems not-ideal, but I do need to have the input and output streams in the same place for copy, and this seemed the only way to keep the fiber.Ctx in the controller only. 

What do you think of passing the fiber.Ctx as `Writer` argument down to storage provider?

Reply:
passing it down also not a good idea. probably the same situation when we share implementation. but why we can't copy data into the `buffer`, pass buffer to `controller`. read and write `buffer` data into `fiber` context right in `controller`?

Reply:
as I mentioned in another comment, this approach seems fine to me — one additional improvement would be to use `SetBodyStreamWriter` so that we send all the right headers to the client and then start streaming the response back (though it's worth checking whether this may already be the case with writing straight to `fiber.Ctx`???)

Reply:
converted to SetBodyStreamWriter (which, unfortunately, doesn't see to let us handle or return error?)